webapp/src/components/modals/create_issue/create_issue.jsx

```diff
@@ -0,0 +1,183 @@
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {Modal} from 'react-bootstrap';
+
+import FormButton from 'components/form_button';
+import GithubRepoSelector from 'components/github_repo_selector';
+import Input from 'components/input';
+
+const initialState = {
+    submitting: false,
+    error: null,
+    repoValue: '',
+    issueTitle: '',
+};
+
+export default class CreateIssueModal extends PureComponent {
+    static propTypes = {
+        close: PropTypes.func.isRequired,
+        create: PropTypes.func.isRequired,
+        post: PropTypes.object,
+        theme: PropTypes.object.isRequired,
+        visible: PropTypes.bool.isRequired,
+    };
+
+    constructor(props) {
+        super(props);
+        this.state = initialState;
+    }
+
+    // handle issue creation after form is populated
+    handleCreate = (e) => {
+        if (e && e.preventDefault) {
+            e.preventDefault();
+        }
+
+        if (!this.state.issueTitle) {
+            return;
+        }
+
+        // const title = this.state.issueTitle;
+        // const description = this.props.post.message;
+        // const repo = this.state.repoValue;
+        // const repoUrlParts = this.state.repoValue.split('/');
+        // const repo = repoUrlParts.pop();
+
+        const issue = {
+            title: this.state.issueTitle,
+            body: this.props.post.message,
+            repo: this.state.repoValue.name,
+        };
+
+        this.setState({submitting: true});
+
+        this.props.create(issue).then((created) => {
+            if (created.error) {
+                this.setState({error: created.error.message, submitting: false});
+                return;
+            }
+            this.handleClose(e);
+        });
+    };
+
+    handleClose = (e) => {
+        if (e && e.preventDefault) {
+            e.preventDefault();
+        }
+        const {close} = this.props;
+        this.setState(initialState, close);
+    };
+
+    handleRepoValueChange = (newValue) => {
+        this.setState({
+            repoValue: newValue,
+        });
+    };
+
+    handleIssueTitleChange = (newValue) => {
+        this.setState({
+            issueTitle: newValue,
+        });
+    };
+
+    render() {
+        const {visible, theme} = this.props;
+        const {error, submitting} = this.state;
+        const style = getStyle(theme);
+
+        if (!visible) {
+            return null;
+        }
+
+        const component = (
+            <div>
+                <GithubRepoSelector
+                    id={'repo'}
+                    onChange={this.handleRepoValueChange}
+                    required={true}
+                    theme={theme}
+                    error={error}
+                    value={this.state.repoValue}
+                />
+                <Input
+                    id={'title'}
+                    label='Title for the Github Issue'
```

Comment:
Thanks @dhadiseputro for the contribution! Propose referencing `GitHub` in all UI texts instead of `Github` :)

Reply:
yea sure, will work on this in a bit.